src/resolve/resolved-dns-rr.h

```diff
@@ -90,6 +91,14 @@ struct DnsTxtItem {
         uint8_t data[];
 };
 
+struct DnsSvcParam {
+        uint16_t key;
+        size_t length;
+        LIST_FIELDS(DnsSvcParam, params);
+        /* Alignment is convinient for reading ip addresses */
+        _alignas_(in_addr_t) uint8_t value[];
```

Comment:
hmm, I think it would be nicer to do this via an anonymous union instead, i.e:

```c
union {
        uint8_t value[];
        struct in_addr value_in_addr;
        struct in6_addr value_in6_addr;
};
```

this means the alignment happens implicitly, and we can conveniently access the fields with correct type without casting, and the compiler understands the aliasing rules here.

Reply:
Damn I thought that was a good idea but the compiler doesn't like it: `src/resolve/resolved-dns-rr.h|99 col 25-30 error| Flexible array member 'value' in a union is not allowed`

Reply:
Can't you just do something like we already do for the `in_addr_union`? That is:

https://github.com/systemd/systemd/blob/81a183800fec3a144b345213066c58431cb7e85b/src/basic/in-addr-util.h#L12-L16

Reply:
This member doesn't have fixed length — it is recorded in the length field copied from the SVCB record. Even in the case of ipv{4,6}hint svc parameters, this field will contain several (length / size) addresses.  So, I guess ideally it would look like

```c
union {
    uint8_t value[];
    struct in_addr value_in_addr[];
    struct in6_addr value_in6_addr[];
}
```

But this is illegal according to gcc.

Reply:
@rpigott Try the `DECLARE_FLEX_ARRAY` array macro and see if that works here

Reply:
Looks like `DECLARE_FLEX_ARRAY` works! Thanks!